cmd/ndm_daemonset/probe/sysfsprobe.go

```diff
@@ -0,0 +1,173 @@
+
+// FillBlockDeviceDetails updates the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+// if those are not populated.
+func (cp *sysfsProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevice) {
+	sysPath := blockDevice.SysPath
+
+	if blockDevice.DeviceAttributes.LogicalBlockSize == 0 {
+		logicalBlockSize, err := readSysFSFileAsInt64(sysPath + "/queue/logical_block_size")
+		if err != nil {
+			klog.Warning("unable to read logical block size", err)
+		} else if logicalBlockSize != 0 {
+			blockDevice.DeviceAttributes.LogicalBlockSize = uint32(logicalBlockSize)
+			klog.Infof("blockdevice path: %s logical block size :%d filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.LogicalBlockSize)
+		}
+	}
+
+	if blockDevice.DeviceAttributes.PhysicalBlockSize == 0 {
+		physicalBlockSize, err := readSysFSFileAsInt64(sysPath + "/queue/physical_block_size")
+		if err != nil {
+			klog.Warning("unable to read physical block size", err)
+		} else if physicalBlockSize != 0 {
+			blockDevice.DeviceAttributes.PhysicalBlockSize = uint32(physicalBlockSize)
+			klog.Infof("blockdevice path: %s physical block size :%d filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.PhysicalBlockSize)
+		}
+	}
+
+	if blockDevice.DeviceAttributes.HardwareSectorSize == 0 {
+		hwSectorSize, err := readSysFSFileAsInt64(sysPath + "/queue/hw_sector_size")
+		if err != nil {
+			klog.Warning("unable to read hardware sector size", err)
+		} else if hwSectorSize != 0 {
+			blockDevice.DeviceAttributes.HardwareSectorSize = uint32(hwSectorSize)
+			klog.Infof("blockdevice path: %s hardware sector size :%d filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.PhysicalBlockSize)
+		}
+	}
+
+	if blockDevice.DeviceAttributes.DriveType == "" {
+		rotational, err := readSysFSFileAsInt64(sysPath + "/queue/rotational")
+		if err != nil {
+			klog.Warning("unable to read rotational value", err)
+		} else {
+			if rotational == 1 {
+				blockDevice.DeviceAttributes.DriveType = "HDD"
+
+			} else if rotational == 0 {
+				blockDevice.DeviceAttributes.DriveType = "SSD"
+			}
+			klog.Infof("blockdevice path: %s drive type :%s filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.DriveType)
+		}
+	}
+
+	if blockDevice.Capacity.Storage == 0 {
+		// The size (/size) entry returns the `nr_sects` field of the block device structure.
+		// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L1267
+		//
+		// Traditionally, in Unix disk size contexts, “sector” or “block” means 512 bytes,
+		// regardless of what the manufacturer of the underlying hardware might call a “sector” or “block”
+		// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L487
+		//
+		// Therefore, to get the capacity of the device it needs to always multiplied with 512
+		blockSize, err := readSysFSFileAsInt64(sysPath + "/size")
```

Comment:
n_: rename the `blockSize` to `numberOfBlocks`?

Reply:
done @kmova .